controllers/loadtest_controller.go

```diff
@@ -175,131 +176,221 @@ func (r *LoadTestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		log.Error(err, "failed to list pods", "namespace", req.Namespace)
 		return ctrl.Result{Requeue: true}, err
 	}
-
 	ownedPods := status.PodsForLoadTest(test, pods.Items)
 
 	previousStatus := test.Status
 	test.Status = status.ForLoadTest(test, ownedPods)
-
 	if err = r.Status().Update(ctx, test); err != nil {
 		log.Error(err, "failed to update test status")
 		return ctrl.Result{Requeue: true}, err
 	}
 
 	missingPods := status.CheckMissingPods(test, ownedPods)
-	builder := podbuilder.New(r.Defaults, test)
-
-	createPod := func(pod *corev1.Pod) (*ctrl.Result, error) {
-		if err = ctrl.SetControllerReference(test, pod, r.Scheme); err != nil {
-			log.Error(err, "could not set controller reference on pod, pod will not be garbage collected", "pod", pod)
-			return &ctrl.Result{}, err
+	if !missingPods.IsEmpty() {
+		if !r.mgr.GetCache().WaitForCacheSync(ctx.Done()) {
+			log.Error(errCacheSync, "could not invalidate the cache which is required to gang schedule")
+			return ctrl.Result{Requeue: true}, errCacheSync
```

Comment:
Return either Result.{Requeue: true}  or an error would result in a requeue behavior. I am wondering if it is possible to just use Result.{Requeue: true} instead of the error. I am still investigating how the error returned would affect the requeueAfter behavior, seems like there is a connection based on my limited observation.

Reply:
I think this is the step you validating cache, wondering how expensive it is

Reply:
@wanlin31 It is certainly possible to just use `ctrl.Result{Requeue: true}`, but I do think this is a problematic situation. I do not see a non-exceptional reason why we would not be able to invalidate the cache. I would like to somehow record these errors.

Reply:
I agree, I think this error is a bit different from "Already exit" and "Update conflict" since the aforementioned two errors are not real errors, one might not want to record them, what we have here could be a real error that needed be recorded.

Reply:
By recording this error, it affects the rate of retrying

Reply:
That's a good point.  If it failed to wait for the cache, the reconcile timeout was reached or their was a problem with the API. I don't know. We do need to retry if it fails b/c otherwise the test will never be scheduled.